aea/mail/base.py

```diff
@@ -340,6 +344,7 @@ def to(self) -> Address:
     @to.setter
     def to(self, to: Address) -> None:
         """Set address of receiver."""
+        enforce(isinstance(to, str), f"To must be string. Found '{type(to)}'")
```

Comment:
may be `isinstance(to, Address)`? i know that is an alias for str

Reply:
same thing in various places

Reply:
yes that could one done eventually when we consistently apply Address